test/CMakeLists.txt

```diff
@@ -84,6 +84,10 @@ if (NOT (MSVC AND BUILD_SHARED_LIBS))
 endif ()
 add_fmt_test(ostream-test)
 add_fmt_test(compile-test)
+add_fmt_test(compile-fp-test HEADER_ONLY)
+if (MSVC)
+  target_compile_options(compile-fp-test PRIVATE /Zc:__cplusplus)
```

Comment:
Why add `/Zc:__cplusplus`?

Reply:
It was added by [a suggestion from](https://github.com/fmtlib/fmt/pull/2426#discussion_r671811386) @DanielaE, to enable `compile-fp-test` (a check there is based on `__cplusplus`) on MSVC C++20.

Reply:
Could you clarify this in a comment?

Reply:
Done